src/proxy.c

```diff
@@ -283,3 +284,22 @@ size_t create_socks_req(proxytype_t type, void *buf, const sockaddr_t *sa) {
 		abort();
 	}
 }
+
+char *exec_proxy_path(void) {
+	if(proxytype == PROXY_EXEC && proxyhost && *proxyhost) {
+		char *command = xstrdup(proxyhost);
+
+		// Find the first non-whitespace token
+		const char *exe = strtok(command, " \t\r\n");
+
+		if(exe) { // If found, remove whitespace prefix and return the path
+			memmove(command, exe, strlen(exe) + 1);
+			return command;
```

Comment:
But `command` might not necessarily be a path to an executable, it can be an arbitrary command line which is passed to `system()`.